### ISSUE TITLE
Connections Request: Update Trakt Collection

### ISSUE THREAD

**Issue by dannyvfilms** [Radarr/Radarr#4521]
There aren't too many plugins I know of that will easily update your Trakt Collection. 

It seems to me that Radarr already knows what movies I have, along with the other metadata involved. An addition to the Connections page to Trakt would be the perfect way to have movies "collected" on Trakt when new files are downloaded.

The Kodi Trakt plugin comes to mind, but that requires using PlexKodiConnect to spoof your Plex library as part of Kodi's library, which then syncs as "Collected" to Trakt if the plugin is installed.

Unfortunately there is no webhook within Plex to sync to Trakt collections, so Radarr/Sonarr are the next step down the chain.

**Comment by DandyDev** [comment]
I second this. I'd love for Radarr to mark a movie as collected in Trakt once it has downloaded a movie.

**Comment by geogolem** [comment]
Yes, I used to use a Plex plugin for this, but I switched to the webhook solution with only scrobbles, and updates watch history.. It doesn't sync the plex library with "collected" list in trakt.. but radarr could do this as you suggest.. Good suggestion!

**Comment by geogolem** [comment]
I see 2 Solutions to this: 

Solution 1) Implement two features that can be enabled with options.
Option 1) Add to Trakt Collection on Download YES/NO
would implement this api call: https://trakt.docs.apiary.io/#reference/sync/add-to-collection/add-items-to-collection

Option 2) Remove from Trakt Collection on Delete YES/NO
would implement this api call: https://trakt.docs.apiary.io/#reference/sync/remove-from-collection/remove-items-from-collection
-----
Solution 2)
Have a new task for sync with Trakt Collection which would use this api call: https://trakt.docs.apiary.io/#reference/sync/get-collection/get-collection

and then issue api calls for any items needing to be added or removed from the trakt collection./

**Comment by Qstick** [comment]
Initial implementation will be just adding to collection via a connection (done in #4841) but eventually I'd like to see your option 2 happen. 

As a second step we need to implement on delete notifications for the second part of option one which shouldn't be terribly difficult. 

But yea ultimate plan is to have 'Integrations' that this would fit nicely into.

**Comment by geogolem** [comment]
this feature was sort of requested in https://github.com/Radarr/Radarr/issues/1362 however that was a different feature request. It appears that that feature request would be resolved by a generalization of the TraktConnection to allow user specified lists and operations. I might look into this in the next few days.

**Comment by rg9400** [comment]
I'll discuss this with you. It's very simple to interact with Trakt, but I think we should plan out exactly how the proper integration would look like.

**Comment by bakerboy448** [comment]
@geogolem closing #1362; cover that here please

**Comment by jt196** [comment]
How is this feature/issue progressing? I'm fairly new to Radarr but it'd be nice to have some automation with the Trakt lists. 

Currently my workflow is:

1. Add movie to Trakt watchlist
2. Radarr imports, downloads film.
3. I watch movie. 
4. I delete movie in Radarr (not in file system as Radarr will re-import it). 
5. I delete movie from Trakt Watchlist. (I may end up going into Trakt and cleaning up my Watchlist - but Trakt doesn't have bulk editing, so it's a slightly painful one by one deletion process). 

Unless I'm missing something here, I can't streamline this any further. It'd be nice to have some automation within Radarr to cover steps 4/5 a bit better. Allowing for more flexibility deleting files, and also some sort of management for moving watched or downloaded films between Trakt lists. The second post in issue [#1362](https://github.com/Radarr/Radarr/issues/1362) covers some solutions but doesn't go into great detail about how they implement this, particularly the removal of the items from the Trakt watchlist. I think there's some reference to the clean library option. 

What I'd like to have is:

1. Add movie to Trakt watchlist
2. Radarr imports, downloads, then moves item to another list
3. I watch movie. (my player - Plex currently - moves the films to another list, this is outlined in #1362)
4. I delete the movie in Radarr, Plex or my file system. 

If there's currently any way of doing this that doesn't involve scripting that'd be great to know. I'm currently in a Docker set up and adding scripts to the system is just a bit more complex than I'm willing to bother with as I'd need to faff around with building Dockerfiles etc and then the hassle of updates. Alternatively a separate container to do this work would also be more hassle than it's worth.

**Comment by geogolem** [comment]
I just use this python tool: https://github.com/Taxel/PlexTraktSync

It seems to work well....

**Comment by aviadlevy** [comment]
What I’d like to suggest for Trakt notifications is handling two cases differently.  
If I delete just the file (but not the movie entry itself), it would be ideal for the movie to be marked as watched. And if I delete the entire movie entry from Radarr, the movie should also be removed from Trakt and no longer appear as collected.
